api.bs

```diff
@@ -1902,26 +1915,146 @@ as close to 2<sup><var>n</var></sup> − 1 as other constraints allow.
+            with [UINT32](https://tc39.es/ecma262/multipage/indexed-collections.html#table-the-typedarray-constructors), |scaledEpsilon|, and `false` (for `isLittleEndian`).
+
+    *   The extension codepoint for [[DAP-EXT#name-requester-website-identity|requester identity]],
+        mapped to the [=isomorphic encode|encoded=] value of |topLevelSite|[1].
+
+1.  Let |reportMetadata| be encoded DAP [`ReportMetadata`](https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-15#section-4.5.2)
+    generated from |reportID|, |time|, and |extensions|.
+
+1.  Let |encryptedInputShares| be an [=list/empty=] [=list=].
+
+1.  For each |share| in |inputShares|,
+    follow the method for encrypting shares
+    described in [Section 4.5.2](https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-15#section-4.5.2):
+
+    1.  Let |pkR| be the public key of the corresponding role from
+        the [=aggregation service=] [HPKE configuration](https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-15#section-4.5.1)
+        obtained for the [=aggregation service=]
+        indicated by |options|.[=validated conversion options/Aggregation Service=].
+
+        <p class=note>The URL for <a enum-value for=PrivateAttributionAggregationProtocol>"dap-15-histogram"</a> is expected to identify the DAP Leader role.
+        Implementations need to obtain HPKE configuration for both Aggregators statically.
+        The HPKE configuration <span class=allow-2119>must not</span> be fetched on demand as the time that takes
+        will leak information to callers of <a method for=Attribution>measureConversion()</a>.
+
+    1.  Let |serverRole| of 2 for the first item (the Leader)
+        and 3 for the second (the Helper role).
+
+    1.  Let |info| be the [=byte sequence=] formed by concatenating:
+        the [=isomorphic encode|encoded=] value of the string "dap-15 input share",
+        a byte with the value 1, and |serverRole|.
+
+    1.  Let |inputShareAAD| be constructed from
+        |taskID|, |reportMetadata|, and |publicShare|,
+        following the structure for [`InputShareAad`](https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-15#section-4.5.2).
+
+    1.  Let |hpke| be an HPKE [[RFC9180]] configuration
+        that is based on the same [HPKE configuration](https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-15#section-4.5.1).
+
+    1.  Let |encryptedShare| be the result of invoking |hpke|.[`Seal<mode_base>()`](https://hpkewg.github.io/hpke/draft-ietf-hpke-hpke.html#section-6.1),
+        passing |pkR|, |info|, |inputShareAAD|, and |share|.
+
+    1.  [=list/Append=] |encryptedShare| to |encryptedInputShares|.
+
+1.  Let |report| be an encoded DAP [`Report`](https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-15#section-4.5.2)
+    generated from |reportMetadata|, |publicShare|, |encryptedInputShares|,
+    and [=aggregation service=] [HPKE configuration](https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-15#section-4.5.1)
```

Comment:
The struct does not take a list of shares, it explicitly names the leader and helper share. Probably better to just have `leaderShare` and `helperShare` as explicit identifiers in that case.